pkg/crc/api/handlers.go

```diff
@@ -213,22 +214,37 @@ func (h *Handler) GetConfig(c *context) error {
 
 	if len(req.Properties) == 0 {
 		allConfigs := h.Config.AllConfigs()
-		configs := make(map[string]interface{})
-		for k, v := range allConfigs {
-			configs[k] = v.Value
+		allSettings := h.Config.AllSettings()
+
+		if len(allConfigs) != len(allSettings) {
+			return fmt.Errorf("settings and configs have different length")
+		}
+
+		configs := make(map[string]client.SettingValue)
+		for _, setting := range allSettings {
+			configs[setting.Name] = client.SettingValue{
+				Value:        allConfigs[setting.Name].Value,
+				IsDefault:    allConfigs[setting.Name].IsDefault,
+				DefaultValue: setting.GetDefaultValue(),
```

Comment:
instead of the `if len(allConfigs) != len(allSettings) {` check, it's probably more robust to do:
```
for _, setting := range allSettings {
	settingValue, settingValueFound := allConfigs[setting.Name]
	if !settingValueFound {
		return fmt.Errorf(...)
	}                       
	configs[setting.Name] = client.SettingValue{
				Value:        settingValue.Value,
				IsDefault:    settingValue.IsDefault,
	...
```